Sources/Flush/Scheme/IntervalFlushScheme.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  IntervalFlushScheme.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/15/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+final class IntervalFlushScheme: FlushScheme {
+
+    private let interval: TimeInterval
+    private var timer: Timer?
+    private weak var delegate: FlushSchemeDelegate?
+
+    init(interval: TimeInterval = 100) {
+        self.interval = interval
+    }
+
+    deinit {
+        stop()
+    }
+
+    func start(delegate: FlushSchemeDelegate) {
+        self.delegate = delegate
+        scheduleTimer()
+    }
+
+    func stop() {
+        invalidTimer()
+    }
+}
+
+extension IntervalFlushScheme {
+
+    fileprivate func invalidTimer() {
+        guard let timer = timer else {
+            return
+        }
+
+        timer.invalidate()
+        self.timer = nil
+    }
+
+    fileprivate func scheduleTimer() {
+        guard timer == nil else {
+            return
+        }
+
+        timer = Timer.scheduledTimer(timeInterval: interval,
```

Comment:
use the `block` version to get rid of the `@objc` ugliness

Reply:
The reason I chose @objc version because of the Timer, which is based-block, only is available on macOS > 10.12.
[Apple Document](https://developer.apple.com/documentation/foundation/timer/2091889-scheduledtimer).
So if I use this Timer, I would like to check availability and fallback. It's unnecessary.